civihr_employee_portal/js/reports.js

```diff
@@ -203,10 +205,83 @@
 
         $(this).find('.pvtCheckContainer p').addClass('chr_custom-checkbox');
       }
+
+      // Altering Employee length of service data to show number of days into
+      // String format of Year Months and days of length of service.
+      // Finding the field called Employee length of service.
+      if($(this).find('h4').text().indexOf("Employee length of service (") >= 0 ) {
+         // Iterating through all the filter values.
+        $(this).find('p.chr_custom-checkbox').each(function(i){
+          // Skip the Select All checkbox from filters.
+          if (!$(this).hasClass("pvtFilterSelectAllWrap")){
+            // Fetching the Employee length of service from filter values.
+            var employee_length_service = $(this).find('span:first').text();
+            // Replacing with format string calling function as parameter.
+            $(this).find('span:first').text(that.formatLengthsOfService(employee_length_service));
```

Comment:
Instead of describing the code line by line, do you think you can make the code itself more self descriptive? Maybe creating some new variables and splitting part of it in some functions?